src/tool.cpp

```diff
@@ -170,7 +171,25 @@ HitParams getHitParams(const ItemGroupList &armor_groups,
 				* armor / 100.0;
 	}
 
-	return HitParams(damage, 0);
+	for (std::map<std::string, ToolGroupCap>::const_iterator
+			i = tp->groupcaps.begin(); i != tp->groupcaps.end(); ++i) {
+		const std::string &name = i->first;
+		const ToolGroupCap &cap = i->second;
+
+		if (cap.uses != 0)
+			result_wear = 1.0 / cap.uses;
+		else
+			result_wear = 0;
+	}
+
+	if (time_from_last_punch < tp->full_punch_interval) {
+		float f = time_from_last_punch / tp->full_punch_interval;
+		//infostream<<"f="<<f<<std::endl;
+		result_wear /= f;
```

Comment:
This code is exactly the same as the wear calculation when you dig, I'll need to debug if that calculation is incorrect or not.